services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt

```diff
@@ -98,22 +108,25 @@ class ApplicationController(
                 ),
             ]
     )
-    fun create(@ValidApplication @RequestBody application: Application): Application {
+    fun create(
+        @RequestParam("luo-hanke") generateHanke: Boolean = false,
+        @ValidApplication @RequestBody application: Application
+    ): Application {
+        logger.info { "Creating new application, param luo-hanke: $generateHanke" }
         val userId = currentUserId()
-        val hankeTunnus = application.hankeTunnus
-        val hankeId = hankeService.getHankeId(hankeTunnus)
 
-        if (
-            hankeId == null ||
-                !permissionService.hasPermission(hankeId, userId, PermissionCode.EDIT_APPLICATIONS)
-        ) {
-            throw HankeNotFoundException(hankeTunnus)
+        if (generateHanke) {
+            return hankeService
+                .generateHankeWithApplication(application, userId)
+                .applications
+                .first()
+                .also { disclosureLogService.saveDisclosureLogsForApplication(it, userId) }
         }
 
-        val createdApplication = service.create(application, userId)
-
-        disclosureLogService.saveDisclosureLogsForApplication(createdApplication, userId)
-        return createdApplication
+        checkApplicationPermission(application.hankeTunnus, userId)
+        return service.create(application, userId).also {
```

Comment:
How is using `also` here clearer than using a variable?

Reply:
Not much difference, I prefer returning the result (and _also_ do the disclosure stuff) straight away. I thought also would be a pretty good option for this.

Reply:
This implementation was reverted to (almost) what it was before this pr (separated endpoints). Is it ok now?